src/components/ShareLink.js

```diff
@@ -60,56 +85,89 @@ export default function ShareLink(props) {
       const bitlySlice = bitlyLink.link.slice(15);
       const displayLink = `https://fshschool.org/FSHOnline/#/share/${bitlySlice}`;
       setLink(displayLink);
-      setOpenShare(true);
       setCopied({ copied: false, copyButton: 'Copy to Clipboard' });
     }
   };
 
-  const handleCloseShare = () => {
-    setOpenShare(false);
-  };
-
-  const handleOpenShareError = () => {
-    setOpenShareError(true);
-  };
-
-  const handleCloseShareError = () => {
-    setOpenShareError(false);
+  const handleGenerateLinkFromGist = () => {
+    const gistId = gistURL.slice(gistURL.lastIndexOf('/') + 1);
+    const link = `https://fshschool.org/FSHOnline/#/gist/${gistId}`;
+    setLink(link);
+    setCopied({ copied: false, copyButton: 'Copy to Clipboard' });
   };
 
   return (
     <>
       <Tooltip title="Share FSH" placement="top" arrow>
         <IconButton name="Share FSH" className={classes.iconButton} onClick={handleOpenShare}>
-          <Link fontSize="small" style={{ transform: 'rotate(-45deg)' }} />
+          <LinkIcon fontSize="small" style={{ transform: 'rotate(-45deg)' }} />
         </IconButton>
       </Tooltip>
       <Dialog open={openShare} onClose={handleCloseShare} aria-labelledby="form-dialog-title" maxWidth="sm" fullWidth>
         <DialogTitle id="form-dialog-title">Share</DialogTitle>
-        <DialogContent>
-          <DialogContentText>Use this link to share your FSH with others!</DialogContentText>
-          <TextareaAutosize
-            id="link"
-            disabled
-            label="Your Link"
-            defaultValue={link}
-            onChange={updateLink}
-            className={classes.textArea}
-          ></TextareaAutosize>
-        </DialogContent>
-        <DialogActions>
-          <CopyToClipboard text={link} onCopy={() => setCopied({ copied: true, copyButton: 'Link Copied' })}>
-            <Button color={copied ? 'secondary' : 'primary'}>{copyButton}</Button>
-          </CopyToClipboard>
-          <Button onClick={handleCloseShare} color="primary">
-            Done
-          </Button>
-        </DialogActions>
+        <Divider orientation="horizontal" style={{ width: '100%' }} />
+        <Grid container alignItems="flex-end">
+          <Grid item xs={6} style={{ borderRight: '1px solid rgba(0,0,0,.12)' }}>
+            <DialogContent>
+              Generate a link directly from FSH Online to share your FSH with others. This option is best for small
+              snippets of FSH.
+            </DialogContent>
+            <DialogActions>
+              <Button onClick={handleGenerateLink} color="primary">
+                Generate Direct Link
+              </Button>
+            </DialogActions>
+          </Grid>
```

Comment:
I'd definitely like your opinion on how difficult this is and if it is worth doing as part of this PR, but I think it might be nice for users to know right away if their FSH is too long to be shared with a direct link. It might be nice to have this section conditionally render what it currently says if the FSH is short enough or render a message that their FSH is too long and should be shared with a Gist instead and then we can disable the Generate Link button here as well.

Reply:
I think this makes sense, it means we will generate more links, but if we weren't bumping into the API limits before, I don't think we will now, and if we do we can always change it up, so sounds good!

Reply:
If bit.ly publishes their max supported URL size, we might be able to detect this before attempting to create the link.  That said, I'm not sure they publish the max size (otherwise we might have already used it by now).  Still... might be worth looking into again.

Reply:
Addressed in https://github.com/FSHSchool/FSHOnline/commit/0780e82215ff0c75c62077db37ffa38348f66519.